Source/Assimil8or/Assimil8or.cpp

```diff
@@ -631,18 +858,44 @@ void Assimil8orPreset::write (juce::File presetFile)
     write (presetFile, presetProperties.getValueTree ());
 }
 
+void Assimil8orPreset::clearVars()
+{
+    curPresetSection = {};
+    channelProperties = {};
+    curChannelSection = {};
+    zoneProperties = {};
+    curZoneSection = {};
+}
+
 // NOTE: still very much under construction
 void Assimil8orPreset::parse (juce::StringArray presetLines)
 {
     jassert (presetProperties.isValid ());
 
     auto scopeDepth { 0 };
+    clearVars ();
+    
+    auto dispatch = [this] (std::map<juce::String, std::function<void ()>> & actions, juce::String assertMessage)
+    {
+        auto paramName = key.upToFirstOccurrenceOf (" ", false, false);
 
-    juce::ValueTree curPresetSection;
-    ChannelProperties channelProperties;
-    juce::ValueTree curChannelSection;
-    ZoneProperties zoneProperties;
-    juce::ValueTree curZoneSection;
+        if (paramName != juce::String (""))
```

Comment:
This is another place where you can use `{}` to indicate a default object of type deduced with type detection.